From 2651caea19e1c320cf23f17a5b9bfcafa6d7d7f7 Mon Sep 17 00:00:00 2001
From: Lucas De Marchi <lucas.demarchi@profusion.mobi>
Date: Thu, 25 Nov 2010 00:17:08 -0200
Subject: [PATCH 5/6] Add VERY_QUIET option

If someone doesn't want to see early print messages, it's sufficient to
compile as below:

 make VERY_QUIET=1

Signed-off-by: Lucas De Marchi <lucas.demarchi@profusion.mobi>
---
 core/Makefile      |    4 ++++
 core/diskstart.inc |    7 +++++++
 core/isolinux.asm  |    4 ++++
 core/pxelinux.asm  |    2 ++
 4 files changed, 17 insertions(+), 0 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index 1330fb9..41f35af 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -69,6 +69,10 @@ ifndef DATE
 DATE    := $(shell sh ../gen-id.sh $(VERSION) $(HEXDATE))
 endif
 
+ifeq ($(VERY_QUIET),1)
+NASMOPT += -DVERY_QUIET=1
+endif
+
 all: $(BTARGET)
 
 kwdhash.gen: keywords genhash.pl
diff --git a/core/diskstart.inc b/core/diskstart.inc
index c0301d4..29bc504 100644
--- a/core/diskstart.inc
+++ b/core/diskstart.inc
@@ -506,11 +506,13 @@ ldlinux_ent:
 ;
 		sti		; In case of broken INT 13h BIOSes
 
+%ifndef VERY_QUIET
 ;
 ; Tell the user we got this far
 ;
 		mov si,syslinux_banner
 		call writestr_early
+%endif
 
 ;
 ; Checksum data thus far
@@ -535,7 +537,9 @@ print_bios:
 		mov byte [getlinsec.jmp+1],(getlinsec_ebios-(getlinsec.jmp+2))
 .cbios:
 		mov [BIOSName],si
+%ifndef VERY_QUIET
 		call writestr_early
+%endif
 
 		section .earlybss
 %define	HAVE_BIOSNAME 1
@@ -835,12 +839,15 @@ all_read:
 		xor ax,ax
 		mov ds,ax
 		mov es,ax
+
+%ifndef VERY_QUIET
 ;
 ; Let the user (and programmer!) know we got this far.  This used to be
 ; in Sector 1, but makes a lot more sense here.
 ;
 		mov si,copyright_str
 		call writestr_early
+%endif
 
 
 ;
diff --git a/core/isolinux.asm b/core/isolinux.asm
index ca8ee3a..8754625 100644
--- a/core/isolinux.asm
+++ b/core/isolinux.asm
@@ -274,6 +274,7 @@ _start_common:
 		mov eax,[si]
 		mov [GetlinsecPtr],eax
 
+%ifndef VERY_QUIET
 		; Show signs of life
 		mov si,syslinux_banner
 		call writestr_early
@@ -283,6 +284,7 @@ _start_common:
 		mov si,[BIOSName]
 %endif
 		call writestr_early
+%endif
 
 		;
 		; Before modifying any memory, get the checksum of bytes
@@ -1129,11 +1131,13 @@ all_read:
 		sub ax,di
 		stosw
 
+%ifndef VERY_QUIET
 ; Tell the user we got this far...
 %ifndef DEBUG_MESSAGES			; Gets messy with debugging on
 		mov si,copyright_str
 		call writestr_early
 %endif
+%endif
 
 ;
 ; Now we're all set to start with our *real* business.	First load the
diff --git a/core/pxelinux.asm b/core/pxelinux.asm
index e8818a6..d73d671 100644
--- a/core/pxelinux.asm
+++ b/core/pxelinux.asm
@@ -208,6 +208,7 @@ adhcp_copy:
 ;
 %include "init.inc"
 
+%ifndef VERY_QUIET
 ;
 ; Tell the user we got this far
 ;
@@ -216,6 +217,7 @@ adhcp_copy:
 
 		mov si,copyright_str
 		call writestr_early
+%endif
 
 ;
 ; do fs initialize
-- 
1.7.3.2

